Mathlib/Analysis/Distribution/ContDiffMapSupportedIn.lean

```diff
@@ -493,6 +507,123 @@ theorem continuous_iff_comp_withOrder {X : Type*} [TopologicalSpace X] (φ : X 
   ext t : 1
   simp [hin, structureMapCLM_apply_withOrder]
 
+variable (E F n K)
+
+/-- The seminorms on the space `𝓓^{n}_{K}(E, F)` given by the sup norm of the iterated derivatives.
+In the scope `Distributions.Seminorm`, we denote them by `N[𝕜; F]_{K, n, i}`
+(or `N[𝕜]_{K, n, i}`), or simply by `N[𝕜; F]_{K, i}` (or `N[𝕜; F]_{K, i}`) when `n = ∞`. -/
+protected noncomputable def seminorm (i : ℕ) : Seminorm 𝕜 𝓓^{n}_{K}(E, F) :=
+  (normSeminorm 𝕜 (E →ᵇ (E [×i]→L[ℝ] F))).comp (structureMapLM 𝕜 n i)
+
+@[inherit_doc ContDiffMapSupportedIn.seminorm]
+scoped[Distributions.Seminorm] notation "N["𝕜"]_{"K","n","i"}" =>
+  ContDiffMapSupportedIn.seminorm 𝕜 _ _ n K i
+
+@[inherit_doc ContDiffMapSupportedIn.seminorm]
+scoped[Distributions.Seminorm] notation "N["𝕜"]_{"K","i"}" =>
+  ContDiffMapSupportedIn.seminorm 𝕜 _ _ ⊤ K i
+
+@[inherit_doc ContDiffMapSupportedIn.seminorm]
+scoped[Distributions.Seminorm] notation "N["𝕜";"F"]_{"K","n","i"}" =>
+  ContDiffMapSupportedIn.seminorm 𝕜 _ F n K i
+
+@[inherit_doc ContDiffMapSupportedIn.seminorm]
+scoped[Distributions.Seminorm] notation "N["𝕜";"F"]_{"K","i"}" =>
+  ContDiffMapSupportedIn.seminorm 𝕜 _ F ⊤ K i
+
+open scoped Distributions.Seminorm
+
+/-- The seminorms on the space `𝓓^{n}_{K}(E, F)` given by sup of the
+`ContDiffMapSupportedIn.seminorm k`for `k ≤ i`. -/
+protected noncomputable def seminorm' (i : ℕ) : Seminorm 𝕜 𝓓^{n}_{K}(E, F) :=
+  (Finset.Iic i).sup (ContDiffMapSupportedIn.seminorm 𝕜 E F n K)
+
+protected theorem withSeminorms :
+    WithSeminorms (ContDiffMapSupportedIn.seminorm 𝕜 E F n K) := by
+  let p : SeminormFamily 𝕜 𝓓^{n}_{K}(E, F) ((_ : ℕ) × Fin 1) :=
+    SeminormFamily.sigma fun i _ ↦
+      (normSeminorm 𝕜 (E →ᵇ (E [×i]→L[ℝ] F))).comp (structureMapLM 𝕜 n i)
+  have : WithSeminorms p :=
+    withSeminorms_iInf fun i ↦ LinearMap.withSeminorms_induced (norm_withSeminorms _ _) _
+  exact this.congr_equiv (Equiv.sigmaUnique _ _).symm
+
+protected theorem withSeminorms' :
+    WithSeminorms (ContDiffMapSupportedIn.seminorm' 𝕜 E F n K) :=
+  (ContDiffMapSupportedIn.withSeminorms 𝕜 E F n K).partial_sups
+
+variable {E F n K}
+
+protected theorem seminorm_apply (i : ℕ) (f : 𝓓^{n}_{K}(E, F)) :
```

Comment:
Don't you want a variant with `N[𝕜]_{K, n, i} f = ‖iteratedFDeriv ℝ i f‖` for `i ≤ n`?

Reply:
The thing is for this to be true you have to bundle `iteratedFDeriv ℝ i f` as a bounded continuous function, which is exactly what `structureMapLM` does. Users can then use [ContDiffMapSupportedIn.structureMapLM_apply_withOrder](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/Distribution/ContDiffMapSupportedIn.html#ContDiffMapSupportedIn.structureMapLM_apply_withOrder).

But really I expect that most of the time people will want to use `seminorm_le_iff[_withOrder]` and `norm_iteratedFDeriv_apply_le_seminorm[_withOrder]`.